### ISSUE TITLE
fasta compile crashes when I try to us to compile a file importing package:file/local.dart

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#32182]
Step to reproduce:

1. Create `pubspec.yaml`.

    ```yaml
    name: test
    dependencies:
      file: any
    ```
2. Run `pub get`
3. Create `main.dart`

    ```dart
    import 'package:file/local.dart';
    void main() {}
    ```
4. Try compiling it with `fasta compile`.

```console
$ pkg/front_end/tool/fasta compile -o $HOME/src/temp/test/main.dart.dill --strong-mode --platform=$HOME/src/dart/sdk/out/ReleaseX64/vm_platform_strong.dill $HOME/src/temp/test/main.dart
Unhandled exception:
type 'QualifiedName' is not a subtype of type 'String' of 'key' where
  QualifiedName is from package:front_end/src/fasta/builder/qualified_name.dart
  String is from dart:core

#0      _HashVMBase&MapMixin&&_LinkedHashMapMixin.[]= (dart:collection-patch/dart:collection/compact_hash.dart:248)
#1      KernelLibraryBuilder.applyMixins.analyzeArguments (package:front_end/src/fasta/kernel/kernel_library_builder.dart:380:36)
#2      KernelLibraryBuilder.applyMixins (package:front_end/src/fasta/kernel/kernel_library_builder.dart:393:25)
#3      KernelLibraryBuilder.addClass (package:front_end/src/fasta/kernel/kernel_library_builder.dart:176:9)
#4      OutlineBuilder.endClassDeclaration (package:front_end/src/fasta/source/outline_builder.dart:404:13)
#5      Parser.parseClass (package:front_end/src/fasta/parser/parser.dart:1733:14)
#6      Parser.parseClassOrNamedMixinApplication (package:front_end/src/fasta/parser/parser.dart:1692:14)

```

**Comment by peter-ahe-google** [comment]
[Proposed fix](https://dart-review.googlesource.com/41820).

**Comment by peter-ahe-google** [comment]
Fixed in 2191139a1a1cc49948f01019e83e569956e58242.
